### PR TITLE
Add option to indicate when files are being processed

### PR DESCRIPTION
**Pull request by jmunson**
| Q                        | A 
| ------------------------ | ---|
| Fixed Issues             | None|
| Patch: Bug Fix?          | No|
| Major: Breaking Change?  |No| 
| Minor: New Feature?      | Yes|
| Tests Added/Pass?        | Existing tests still pass, did not add new tests to cover this options output|
| Spec Compliancy?         | doesn't change anything spec related|
| License                  | MIT|
| Doc PR                   | No|
| Any Dependency Changes?  | No|

I added the flag -n or --noisy to add output when files are being processed in watch mode.
This should make it easier to integrate with external debugging tools like [VSCode](https://code.visualstudio.com/docs/editor/tasks#_background-watching-tasks) that can be configured to wait for a background watching task to finish processing the files before it attempts to run or debug them.

### CODE REVIEW

**Comment by ORESoftware** [comment]
+1

**Comment by ORESoftware** [comment]
@jmunson  in this thread would you mind describing what stdout/stderr gets written? I am interested to see if this PR solves my issue https://github.com/babel/babel/issues/6590

**Comment by jmunson** [comment]
@ORESoftware I have the extra output defined as
```js
const startMsg = "Detected change to watched files. Rebuilding..";
const endMsg = "Rebuild complete. Waiting for changes.";
```

The startMsg gets output before it starts processing, and endMsg gets output when it is done. It's all still done using console.log() so it should all be going to stdout.

I should also mention this is implemented with a setTimeout that checks every 250ms to make sure it isn't still processing, so the output line may show up a quarter of a second later. This is to make sure that if you're bulk processing a lot of files, you should only see one instance of the startMsg and endMsg.

**Comment by ORESoftware** [comment]
@jmunson  could you clarify why you need to poll every 250ms? Why do we not know the exact time it's done?

**Comment by jmunson** [comment]
@ORESoftware I wrote it a while ago so I may be misremembering, but originally I did it more directly and without the timeout.

What happened was if you were to add or edit multiple files while watch mode was open,  each add/change would be processed independently, and each one would get a startMsg and endMsg wrapping it.  This was technically accurate, but could be spammy, and especially in the case of something like an editor waiting on this task to enter a 'ready' state, it would cause it to quickly toggle between unready and ready for each file.

Changing it to involve a timer instead makes it so that when the first change comes in, it sees the timer isnt already running and it outputs the startmsg,  then the next event comes in, sees the timer is running and does NOT output the startmsg again. It then clears the old timer and sets a new one.  The endmsg is then printed when the timer is finally allowed to expire -- that is to say, when there isnt another new event coming in quick enough.


TL;DR: It's so that you only get one startmsg and endmsg when processing multiple files, instead of one per file

**Comment by jmunson** [comment]
Also I noticed there is now a conflict here. Let me know if you need me to rebase this.

**Comment by ORESoftware** [comment]
Well it should be nice if after each file is started/finished transpiling, it wrote the filepath to stdout?

**Comment by FelixKuehl** [comment]
May I ask, what the status of this is? Would be very useful to get proper stdout indications.

**Comment by jmunson** [comment]
@FelixKuehl I haven't touched it since writing it and it looks like it has fallen behind vs master. I never did get much feedback from the babel team so I do not know if there is any interest in me updating it.

**Comment by JCB-K** [comment]
My team would love this feature as well, and we can help out if needed. Since this PR some logging has actually been added to master, but it only works on the initial run. A merge of the 2 logging solutions would be good, since the one on master logs the amount of files processed.

**Comment by FelixKuehl** [comment]
I would be happy to help if needed. Having no proper stdout from the babel watcher after first compilation is really confusing and misleading. Would be a really helpful feature to have.
